### PR TITLE
fix: Don't crash when DiContextConsumer first receives `undefined`

### PR DESCRIPTION
**Pull request by Nokel81**

### CODE REVIEW

File: packages/injectable/react/src/withInjectables/withInjectables.test.jsx [hunk 1]
```diff
@@ -166,6 +166,33 @@ describe('withInjectables', () => {
     expect(rendered.baseElement).toMatchSnapshot();
   });
 
+  it('given component and sync dependencies, when rendered without provider, renders to null', () => {
```
Comment (Iku-turso) [inline packages/injectable/react/src/withInjectables/withInjectables.test.jsx:169]: Would it be more justifiable to throw an explanatory error about missing `di` in context? I mean, that should always be a case of misuse, right?

Comment (Iku-turso) [reply packages/injectable/react/src/withInjectables/withInjectables.test.jsx:169]: Or are there justified, yet unfair, reasons for this to momentarily be the case? Why do we discover it only now?

Comment (Nokel81) [reply packages/injectable/react/src/withInjectables/withInjectables.test.jsx:169]: I don't know, why we are just finding it now.

I know that you seem to be abusing the API according to the documentation and that context consumers should be written in a way that is resliant against being rendered outside of a provider.

Comment (Iku-turso) [reply packages/injectable/react/src/withInjectables/withInjectables.test.jsx:169]: > I don't know, why we are just finding it now.

I consider it likely that the misdesign to cause missing `di` is somewhere else, and to silently render `null` in response is a harmful hack around that. Hiding from unexpected circumstances is a questionable policy in software, as it guides towards not understanding what is the real root of problem.

> I know that you seem to be abusing the API according to the documentation and that context consumers should be written in a way that is resliant against being rendered outside of a provider.

Do you have link to API-documentation where the requirement for said resiliency is explained? Common sense would indicate that missing `di`
> should always be a case of misuse, right?

Comment (Iku-turso) [reply packages/injectable/react/src/withInjectables/withInjectables.test.jsx:169]: > I consider it likely that the misdesign to cause missing `di` is somewhere else...

This turned out to be the case, closing PR :)

**Pull request closed without merging**